File: backend/src/main/java/com/kerflowapp/kerflow/api/prospects/domain/ProspectMessageDto.java
package com.kerflowapp.kerflow.api.prospects.domain;

import com.kerflowapp.kerflow.domain.enums.MessageChannel;
import com.kerflowapp.kerflow.domain.enums.MessageDirection;
import com.kerflowapp.kerflow.domain.enums.MessageStatus;
import lombok.Builder;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.UUID;

@Builder
public record ProspectMessageDto(
    UUID id,
    MessageDirection direction,
    MessageChannel channel,
    MessageStatus status,
    String subject,
    String body,
    String generatedBy,
    Instant sentAt,
    Instant receivedAt,
    LocalDateTime creationDate
) {
}

File: frontend/src/api/dtos/prospect-message.dto.ts
export type MessageDirection = 'OUTBOUND' | 'INBOUND';

/** Outbound only: DRAFT until the message is actually sent. Null for inbound. */
export type MessageStatus = 'DRAFT' | 'SENT';

/** Support of the exchange. Null on messages recorded before channels existed → read as EMAIL. */
export type MessageChannel = 'EMAIL' | 'PHONE' | 'SMS' | 'LINKEDIN' | 'MEETING' | 'OTHER';

export interface ProspectMessageDto {
	id: string;
	direction: MessageDirection;
	channel?: MessageChannel | null;
	status?: MessageStatus | null;
	subject?: string | null;
	body: string;
	/** "mcp-agent" when written by an MCP client; null when typed by the user */
	generatedBy?: string | null;
	sentAt?: string | null;
	receivedAt?: string | null;
	creationDate: string;
}

export interface CreateProspectMessageDto {
	/** Omitted → INBOUND */
	direction?: MessageDirection;
	/** Omitted → EMAIL */
	channel?: MessageChannel;
	/** Outbound only, ignored on inbound. Omitted → DRAFT */
	status?: MessageStatus;
	subject?: string;
	body: string;
	/** ISO instant of the exchange. Omitted → now */
	occurredAt?: string;
}
